com.microsoft.mrtk.uxcore/SeeItSayIt/SeeItSayItLabelCreator.cs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using TMPro;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UX
+{
+    [RequireComponent(typeof(PressableButton))]
+    [AddComponentMenu("MRTK/UX/See It Say It Label")]
+    public class SeeItSayItLabelCreator : MonoBehaviour
+    {
+        [SerializeField]
+        [Tooltip("The prefab for the see-it say-it label to be generated.")]
+        private GameObject seeItSayItPrefab;
+
+        /// <summary>
+        /// The prefab for the see-it say-it label to be generated.
+        /// </summary>
+        public GameObject SeeItSayItPrefab
+        {
+            get => seeItSayItPrefab;
+            set => seeItSayItPrefab = value;
+        }
+
+        [SerializeField]
+        [Tooltip("The Transform that the label will be positioned off of. If this is a Canvas button, this should be a RectTransform.")]
+        private Transform positionControl;
+
+        /// <summary>
+        /// The Transform that the label will be positioned off of. If this is a Canvas button, this should be a RectTransform.
+        /// </summary>
+        public Transform PositionControl
+        {
+            get => positionControl;
+            set => positionControl = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Is this a Canvas button?")]
+        private bool isCanvas;
+
+        /// <summary>
+        /// Is this a Canvas button?
+        /// </summary>
+        public bool IsCanvas
+        {
+            get => isCanvas;
+            set => isCanvas = value;
+        }
+
+        [SerializeField]
+        [Tooltip("The offset to bring the label forward by.")]
+        private float forwardOffset;
+
+        /// <summary>
+        /// The offset to bring the button forward by.
+        /// </summary>
+        public float ForwardOffset
+        {
+            get => forwardOffset;
+            set => forwardOffset = value;
+        }
+
+        [SerializeField]
+        [Tooltip("The offset at the bottom of the button.")]
+        private float bottomOffset;
+
+        /// <summary>
+        /// The offset at the bottom of the button.
+        /// </summary>
+        public float BottomOffset
+        {
+            get => bottomOffset;
+            set => bottomOffset = value;
+        }
+
+        private void Start()
+        {
+            // Check if voice commands are enabled for this button
+            PressableButton pressablebutton = gameObject.GetComponent<PressableButton>();
+            if (pressablebutton != null && pressablebutton.AllowSelectByVoice)
+            {
+                // Check if input and speech packages are present
+#if MRTK_INPUT_PRESENT && MRTK_SPEECH_PRESENT
+                GameObject label = Instantiate(SeeItSayItPrefab, transform, false);
```

Comment:
> Adds a SeeItSayItLabelCreator script which creates SeeItSayIt labels at runtime

Spawning these at runtime seems like it could easily become a performance concern. Some scenes may have large numbers of speech-enabled buttons, and I imagine there would be a stall at app launch as these labels are all spawned at once.

Is there any way to make this an embedded edit-time component that only displays if speech is present and enabled (and otherwise disables itself or something)?

Reply:
Yes, I could add a SeeItSayIt label to the button prefabs and change the script so that it enables the label instead of creating it. Is that reasonable?

Reply:
Closing to create a new PR